### ISSUE TITLE
Extremely Long Generation time and an Error

### ISSUE THREAD

**Issue by GalenMarek14** [edwko/OuteTTS#26]
I tried the default voices and also voice cloning in the Gradio demo with a 14-second input audio and transcription. It starts generating with an error and continues, but very slowly. A 14-second output takes around 3 minutes (on a Windows 11 laptop with a 4090 GPU), whether I use normal voices or voice cloning.

The error:

> The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
> Setting `pad_token_id` to `eos_token_id`:None for open-end generation.
> The attention mask is not set and cannot be inferred from input because pad token is same as eos token. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.

Is it supposed to take this long, or is the slow performance related to this error? How can I resolve this issue?

Thanks.

**Comment by edwko** [comment]
The performance you're experiencing seems unusually slow, especially given your hardware. I tested this on a 4090 GPU on Linux, and it took about 20 seconds for an 11 second audio clip using bfloat16 and flash_attention_2.

You can try these settings:

```python
model_config = outetts.HFModelConfig_v1(
    model_path="OuteAI/OuteTTS-0.2-500M",
    language="en",  # Supported languages: en, zh, ja, ko
    dtype=torch.bfloat16,
    additional_model_config={
        'attn_implementation': "flash_attention_2"
    }
)
```

The error about `attention_mask` and `pad_token_id` can be ignored. It just means the tokenizer defaults to using the eos token as the padding token, this won't affect the performance.

**Comment by GalenMarek14** [comment]
I got it, thanks. It seems that installing flash_attn from wheels changed the PyTorch version, so I just reinstalled PyTorch and it opened. It's faster now; default voices generate output that is 2-2.5 times the duration of the output, and voice cloning takes around 5-6 times the output duration.

**Comment by edwko** [comment]
> I got it, thanks. It seems that installing flash_attn from wheels changed the PyTorch version, so I just reinstalled PyTorch and it opened. It's faster now; default voices generate output that is 2-2.5 times the duration of the output, and voice cloning takes around 5-6 times the output duration.

Glad you got it resolved! If you're looking for better performance, I’d recommend using the `llama.cpp` implementation with `GGUF` quantizations.
